GDJS/Runtime/scenestack.ts

```diff
@@ -91,13 +100,33 @@ namespace gdjs {
      * Pause the scene currently being played and start the new scene that is specified.
      * If `externalLayoutName` is set, also instantiate the objects from this external layout.
      */
-    push(newSceneName: string, externalLayoutName?: string): gdjs.RuntimeScene {
+    push(
+      newSceneName: string,
+      externalLayoutName?: string
+    ): gdjs.RuntimeScene | null {
       // Tell the scene it's being paused
       const currentScene = this._stack[this._stack.length - 1];
       if (currentScene) {
         currentScene.onPause();
       }
 
+      if (this._runtimeGame.areLayoutAssetsReady(newSceneName)) {
```

Comment:
Also because it's public interface, would be worth naming it `areSceneAssetsReady`